src/components/encoding-pane/property-editor-schema.ts

```diff
@@ -232,27 +223,52 @@ function getSupportedScaleTypes(shelfId: ShelfId, fieldDef: ShelfFieldDef): stri
         return ["time", "utc", "sequential"];
       } else if (shelfId.channel === Channel.SIZE) {
         return ["time", "utc"];
+      } else {
+        return [];
       }
     default:
       return [];
   }
 }
 
-/*
-*  NOTE: factory method where propertyKey follows naming convention: prop_nestedProp
-*  Example: {prop: axis, nestedProp: orient} translates to "axis_orient"
-*/
-function generateSchema(prop: string, nestedProp: string, uiSchemaItem: UISchemaItem, propTab: string,
-                        schemaProp: SchemaProperty): PropertyEditorSchema {
-  const propertyKey = nestedProp ? prop + '_' + nestedProp : prop;
+function generateSelectSchema(propKey: string, enumVals: string[], baseUISchema: SchemaProperty) {
+  const schemaProp = {
```

Comment:
Also no need to declare this `schemaProp` thing ... given it is only used once and has a bit confusing name anyway.